From dac644ebaabd97af843d2253c0ec3cef0eb801b5 Mon Sep 17 00:00:00 2001
From: Robby Cai <r63905@freescale.com>
Date: Wed, 3 Dec 2014 11:24:32 +0800
Subject: [PATCH] MLK-9958 ARM: dts: imx6sx-19x19-arm2: add new DTS file to
 enable LCDIF1

Due to the CSI and LCDIF1 shares the same pin MX6SX_PAD_LCD1_ENABLE,
we need to disable CSI.

Signed-off-by: Robby Cai <r63905@freescale.com>
---
 arch/arm/boot/dts/Makefile                     |  1 +
 arch/arm/boot/dts/imx6sx-19x19-arm2-lcdif1.dts | 19 +++++++++++++++++++
 2 files changed, 20 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 97be1c1..cd6caeb 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -193,6 +193,7 @@ dtb-$(CONFIG_ARCH_MXC) += \
 	imx6sx-17x17-arm2-mlb.dtb \
 	imx6sx-19x19-arm2.dtb \
 	imx6sx-19x19-arm2-csi.dtb \
+	imx6sx-19x19-arm2-lcdif1.dtb \
 	imx6sx-19x19-arm2-gpmi-weim.dtb \
 	imx6sx-sdb.dtb \
 	imx6sx-sabreauto.dtb \
diff --git a/arch/arm/boot/dts/imx6sx-19x19-arm2-lcdif1.dts b/arch/arm/boot/dts/imx6sx-19x19-arm2-lcdif1.dts
new file mode 100644
index 0000000..cf8cb57
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sx-19x19-arm2-lcdif1.dts
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sx-19x19-arm2.dts"
+
+&i2c2 {
+	ov564x: ov564x@3c {
+		status = "disabled";
+	};
+};
+
+&lcdif1 {
+	status = "okay";
+};
-- 
1.8.0

